packages/store/addon/-private/system/model/internal-model.ts

```diff
@@ -449,7 +468,8 @@ export default class InternalModel {
     this._doNotDestroy = false;
 
     if (this._record) {
-      this._record.destroy();
+      this.store.teardownRecord(this._record);
+      //this._record.destroy();
```

Comment:
ff removal since `teardownRecord` is sub-classable and thus this behavior could be done away with